### PR TITLE
Removed use of edm::EDFilter & edm::EDProducer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This finished the migration away from the deprecated class.

#### PR validation:

- Doing a 'git grep --cached interface/EDFilter.h' turns up no hits. 
- The core framework packages compile.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
hold

This is more for 13_0_X.

**Comment by Dr15Jones** [comment]
The 'header inconsistency' error is because the test is loading `FWCore/Framework/interface/EDFilter.h` and sees the `#error`. So this is actually a success, not a failure.

**Comment by Dr15Jones** [comment]
please test with #39750

**Comment by makortel** [comment]
unhold

**Comment by makortel** [comment]
@Dr15Jones The IBs should be clean from legacy EDAnalyzers as well starting from CMSSW_13_0_X_2023-01-08-0000. Could you (rebase and) update this PR to remove `edm::EDAnalyzer` as well?

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
> @Dr15Jones The IBs should be clean from legacy EDAnalyzers as well starting from CMSSW_13_0_X_2023-01-08-0000. Could you (rebase and) update this PR to remove `edm::EDAnalyzer` as well?

Just for the record, we just agreed that if this PR's tests succeed, we'll go with this PR as it is, and open a new PR for the legacy EDAnalyzers

**Comment by Dr15Jones** [comment]
The 'header inconsistency' is a false alarm. It comes from the `#error` in the deprecated headers

```
src/FWCore/Framework/interface/EDFilter.h:4:2: error: #error "The use of edm::EDFilter has been removed. Please stop use of this header and switch module to a thread-friendly type."
    4 | #error \
      |  ^~~~~
gmake: *** [config/SCRAM/GMake/Makefile.rules:1722: tmp/el8_amd64_gcc11/src/FWCore/Framework/src/FWCoreFramework/EDFilter.h.chk_header] Error 1

src/FWCore/Framework/interface/EDProducer.h:4:2: error: #error "The use of edm::EDProducer has been removed. Please stop use of this header and switch module to a thread-friendly type."
    4 | #error \
      |  ^~~~~

```

File: FWCore/ParameterSet/interface/ParameterSetDescriptionFillerBase.h [hunk 1]
```diff
@@ -106,7 +105,6 @@ namespace edm {
 
     static const std::string& extendedBaseType(EDAnalyzer const*) { return kExtendedBaseForEDAnalyzer; }
     static const std::string& extendedBaseType(EDProducer const*) { return kExtendedBaseForEDProducer; }
```
Comment (makortel) [inline FWCore/ParameterSet/interface/ParameterSetDescriptionFillerBase.h:107]: I think this PR should have removed the `EDProducer` overload as well. On the other hand, I suspect #40466 could already be in (mild, touching adjacent lines) conflict with this PR, so maybe it is easier to leave the cleanup to #40466 or later PR.

**Comment by makortel** [comment]
+1

Header consistency check failure is intentional, and further cleanup will be done in later PR(s).

**Comment by perrotta** [comment]
> The 'header inconsistency' is a false alarm. It comes from the `#error` in the deprecated headers

@Dr15Jones , does it mean that we must live with it in the IBs?

**Comment by makortel** [comment]
> > The 'header inconsistency' is a false alarm. It comes from the `#error` in the deprecated headers
> 
> does it mean that we must live with it in the IBs?

@smuzaffar Would it be easy to explicitly ignore the header consistency check for specific headers?

An alternative would be to remove the headers, whose main consequence is that a user forward-porting their (legacy) code would get a (much) less clear error message from the compiler.

**Comment by smuzaffar** [comment]
@makortel , the failed header are only used by [a]. So why not just move these header in to `FWCore/Framework/src` ?

[a]
```
FWCore/Framework/interface/EDProducer.h: FWCore/Framework/src/EDProducer.cc FWCore/Framework/src/WorkerT.cc 
FWCore/Framework/interface/EDFilter.h: FWCore/Framework/src/EDFilter.cc FWCore/Framework/src/WorkerT.cc 
```

**Comment by smuzaffar** [comment]
ah its forward porting code .... for those they are going to get an error any way so I would suggest to just delete these headers

**Comment by Dr15Jones** [comment]
@smuzaffar 

>the failed header are only used by [a]. So why not just move these header in to FWCore/Framework/src ?

The point isn't that CMSSW depends on those headers (it shouldn't any more) it is that some people who have their own copy repositories of CMSSW still depend on that header and if they try to rebase their code onto a more recent CMSSW release we'd like them to get a better error message (than just 'header file missing').

**Comment by perrotta** [comment]
Thank you all.
I think that an explanatory message, as the one designed by Chris, is welcome because of the user code that may still depend on them, as already pointed out.
My (naive) question was simply: will the IB's get broken by this PR, or will we get filled by several of those header consistency errors that can hide real ones?
If that is not the case, I think the PR can go even as it is.

**Comment by smuzaffar** [comment]
@perrotta , IB itself should not be broken (no one in cmssw master is using these headers). Only the IB header consistency check will show errors about these files. I am preparing a cmssw-config PR to ignore header checks for these.

**Comment by smuzaffar** [comment]
https://github.com/cms-sw/cmsdist/pull/8242 should allow to ignore selected files for header consistency checks

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Comment by smuzaffar** [comment]
https://github.com/cms-sw/cmssw/pull/40474 should fix the header consistency test

**Comment by perrotta** [comment]
> #40474 should fix the header consistency test

Thank you @smuzaffar !

**Pull request merged**